lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1299,26 +1299,61 @@ getNetworkInformation (_block0, bp, st) nl = do
                 }
         }
   where
-    sp :: W.SlotParameters
-    sp = W.SlotParameters
-        (bp ^. #getEpochLength)
-        (bp ^. #getSlotLength)
-        (bp ^. #getGenesisBlockDate)
-        (bp ^. #getActiveSlotCoefficient)
+    sp = toSlotParameters bp
 
     -- Unsafe constructor for the next epoch. Chances to reach the last epoch
     -- are quite unlikely in this context :)
     unsafeEpochSucc :: HasCallStack => W.EpochNo -> W.EpochNo
     unsafeEpochSucc = fromMaybe bomb . W.epochSucc
       where bomb = error "reached final epoch of the Blockchain!?"
 
+toSlotParameters :: BlockchainParameters -> W.SlotParameters
+toSlotParameters bp = W.SlotParameters
```

Comment:
Already exists as https://github.com/input-output-hk/cardano-wallet/blob/3be4db2f27b20ed2e66dfdbaef2e997801b7bd7c/lib/core/src/Cardano/Wallet/Primitive/Types.hs#L1135-L1141

Reply:
ok, thanks - I will definitely use this from Primitive.Types